### PR TITLE
[Enhancement] Support adaptive dop for broker load & insert into

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

When the StarRocks is initially deployed, we set `ENABLE_ADAPTIVE_SINK_DOP=true` so that the `broker load` and `insert into` is automatically configured as the best performance configuration. 
If StarRocks is upgraded from an old version, the `ENABLE_ADAPTIVE_SINK_DOP=false`, the original configuration is retained to avoid system stability problems caused by change load concurrency.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [x] This pr will affect users' behaviors
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by stephen-shelby — APPROVED** [review]

**Comment by meegoo** [comment]
run starrocks_be_unittest

**Comment by meegoo** [comment]
run starrocks_be_unittest

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 1]
```diff
@@ -281,6 +281,7 @@ public class SessionVariable implements Serializable, Writable, Cloneable {
     public static final String HIVE_PARTITION_STATS_SAMPLE_SIZE = "hive_partition_stats_sample_size";
 
     public static final String PIPELINE_SINK_DOP = "pipeline_sink_dop";
```
Comment (howrocks) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:283]: `PIPELINE_SINK_DOP`  is not needed anymore，should be removed.

**Pull request merged**